vaccine_feed_ingest/runners/md/arcgis/normalize.py

```diff
@@ -0,0 +1,435 @@
+            return _parse_days_and_hours(raw.replace("noon", "12pm"))
+        raise RuntimeError(f"Unable to parse time string '{raw}'")
+    hours = tuple(
+        _normalize_hours(h) for h in re.split("-|to", time_match.group("time"))
+    )
+    days = _normalize_days(raw.replace(time_match.group("time"), "").strip())
+    return days, hours
+
+
+def _get_opening_hours(site: dict) -> Optional[List[schema.OpenHour]]:
+    """
+    "24 Hours"
+    "Mon - Fri 7 a.m. - 11 a.m."
+    "Mon - Fri 7 a.m. - 7 p.m."
+    "Mon - Fri 7 am - 7 pm"
+    "Mon - Fri 7am - 7 pm"
+    "Mon - Fri 8 a.m. - 4:30 p.m."
+    "Mon - Fri 8 a.m. - 5 p.m. "
+    "Mon - Fri 8 a.m. - 5 p.m. covid testing Tues. 9 a.m. - 3 p.m. covid vaccine Wed and Fri 9 a.m. - 3 p.m."
+    "Mon - Fri 8 a.m. - 5 p.m. covid testing Wed 9 a.m. - 3 p.m. this week.  Tues. 9 a.m. - 3 p.m. next week. covid vaccine Mon, Wed, Thurs, Fri and Saturday this week"
+    "Mon - Fri 8 a.m. - 5 p.m."
+    "Mon - Fri 8:30 a.m. - 4:30 p.m.covid hotline Mon - Sat 8:30 a.m. - 4:30 p.m. and Sun 8:30 a.m. - 2 p.m."
+    "Mon - Fri 8:30 a.m. - 6:30 p.m. Sat. 9 a.m. - 1 p.m."
+    "Mon, Tues, Thurs, Fri: 7:30am - 3pm, Wed: 3pm - 7pm"
+    "Mon- Fri 7 a.m. - 11 a.m. and 2 PM-4 PM "
+    "Mon- Fri 7 a.m. - 11 a.m."
+    "Sun 10am - 4pm, Mon - Fri 10am - 7pm, Sat 10am - 5pm"
+    "Sun 10am - 5pm, Mon - Fri 8am - 9pm, Sat 9am - 6pm"
+    "Sun 10am - 5pm, Mon - Fri 9am - 9pm, Sat 9am - 6pm"
+    "Sun 10am - 6pm, Mon - Fri 8am - 9pm, Sat 9am - 6pm"
+    "Sun 9am - 5pm, Mon - Fri 8am - 10pm, Sat 9am - 7pm"
+    "Sun 9am - 5pm, Mon - Fri 8am - 9pm, Sat 9am - 7pm"
+    "Varies"
+    "Wed - Sat 10am to 6pm"
+    "A limited number of walk-up appointments are available Monday through Saturday, from 10 a.m. - 12 p.m."
+    "Limited walk-up, no appointment slots available Mon - Tues 9 a.m. - 1 p.m., Wed - Thurs 2 p.m. - 6 p.m., Fri - Sat 11 a.m. - 3 p.m."
+    "Limited walk-up, no appointment slots available Weds -Sun 10 a.m. - 2 p.m."
+    "Limited walk-up, no appointment slots available each day 10 a.m. - 2 p.m."
+    null
+    """
+    all_hours = []
+
+    # TODO: try to merge operationalhours and WalkUpHours ranges that overlap?
+    for attr in ("operationalhours", "WalkUpHours"):
+        raw = site["attributes"][attr]
+        if not raw:
+            continue
+        processed = raw.lower().strip()
+
+        if processed == "24 hours":
+            all_hours.extend(
+                [
+                    schema.OpenHour(day=d, opens="00:00", closes="23:59")
+                    for d in ALL_DAYS
+                ]
+            )
+            continue
+
+        for piece in _pieces(_strip_flavor_text(processed)):
+            days, hours = _parse_days_and_hours(piece)
+            all_hours.extend(
+                [schema.OpenHour(day=d, opens=hours[0], closes=hours[1]) for d in days]
+            )
+
+    return all_hours or None
+
+
+def _get_notes(site: dict) -> Optional[List[str]]:
```

Comment:
Should we add the stripped `flavor text` from above to the notes?

Reply:
Yeah, I waffled on this one.  The trick is there's no feedback mechanism right now from the opening hours parsing to indicate it'd be needed, and it seemed like including the full string for all sites' hours might be overkill.  Should I err on the side of overkill?

Reply:
Though with fresh eyes, we can probably check for `/limited (number of )?walk-up/` in the `WalkUpHours` fields to get just those ones.  Lemme do that.

Reply:
Good catch / addition!